### PR TITLE
Using information schema

### LINKED ISSUE #113: [CT-162] Upgrade from the __tables__ construct to the information_schema.tables construct
### Describe the feature
The use of the [project_id].[dataset_id].__tables__ has been deprecated in favor of [project_id].[dataset_id].information_schema.tables. This is relevant because it is not possible to access the former with metadata only permissions (it requires getData permissions). This would allow secure doc generation and schema-only tests to be run in a lower privilege environment.

### Describe alternatives you've considered
- You can grant the service account dataViewer access to all datasets managed by dbt and all datasets imported as sources. But this could be a massive over privilege.
- You can individually grant dataViewer to these specific tables but that becomes a huge headache as you end up with a grant for every new dataset and always need to keep it up to date or you'll end up with a failure.

### Additional context
None.

### Who will this benefit?
Anyone in high-security or high-compliance environments who want to utilize external dbt actors for certain tasks (github actions, for instance.)

### Are you interested in contributing this feature?
Sure!

### PR DESCRIPTION
**Pull request by hassan-mention-me**
resolves #113 https://github.com/dbt-labs/dbt-bigquery/issues/113

Using `information_schema` instead of __TABLES__ to fetch catalog data, which limits access level needed to generate DBT docs.
### Checklist

- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] I have updated the `CHANGELOG.md` and added information about my change to the "dbt-bigquery next" section.

### CODE REVIEW

**Comment by dansun117** [comment]
currently running into the same issue and found this PR, looks like there are conflicts in this branch.

File: tests/unit/test_bigquery_adapter.py [hunk 1]
```diff
@@ -473,8 +473,17 @@ def test_replace(self):
         relation = BigQueryRelation.from_dict(kwargs)
         info_schema = relation.information_schema()
 
-        tables_schema = info_schema.replace(information_schema_view='__TABLES__')
-        assert tables_schema.information_schema_view == '__TABLES__'
+        tables_schema = info_schema.replace(information_schema_view='INFORMATION_SCHEMA.TABLES')
```
Comment (sungchun12) [inline tests/unit/test_bigquery_adapter.py:476]: `TABLE_STORAGE` is also needed for multi-region BigQuery setups as illustrated in this macro workaround this PR is solving for natively: https://github.com/GeneralMills/gmi_common_dbt_utils/blob/main/macros/bq_catalog.sql#L51

**Comment by sungchun12** [comment]
@hassan-mention-me Any updates on this?

**Comment by ohiosonia** [comment]
This is something General Mills is also interested in seeing

**Comment by colin-rogers-dbt** [comment]
Closing this in favor of getting #364 over the line

**Pull request closed without merging**